include/nbl/asset/filters/CBlitImageFilter.h

```diff
@@ -431,10 +431,10 @@ class CBlitImageFilter :
 					void* const dstPix = outData+writeBlockArrayOffset;
 					const core::vectorSIMDu32 localOutPos = writeBlockPos - outOffsetLayer;
 
-					value_t sample[MaxChannels];
+					value_t sample[ChannelCount];
 					const size_t offset = IImage::SBufferCopy::getLocalByteOffset(localOutPos, intermediateStrides[axis]);
 					const auto* first = intermediateStorage[axis]+offset;
-					std::copy(first,first+MaxChannels,sample);
+					std::copy(first,first+ChannelCount,sample);
 
 					sample[alphaChannel] *= coverageScale;
```

Comment:
do you have anything in `validate()` that checks/ensures `alphaChannel<=ChannelCount` ?

Reply:
in base_t validation there is 

```cpp
if (state->alphaChannel>=4)
	return false;
```

but cannot see that check above anywhere so adding it

Reply:
resolved in following commit